File: front/mobile/Android/PasswordStorage/mobile/src/main/java/com/ndu/passwordstorage/data/PasswordDatabase.java
package com.ndu.passwordstorage.data;

import com.ndu.passwordstorage.model.PasswordEntry;

import java.util.List;

public interface PasswordDatabase {
    List<PasswordEntry> select();
    boolean insert(PasswordEntry passwordEntry);
    boolean update(PasswordEntry passwordEntry);
    boolean delete(PasswordEntry passwordEntry);
    void closeDatabase();
}

File: front/mobile/Android/PasswordStorage/mobile/src/main/java/com/ndu/passwordstorage/data/PasswordDatabase.kt
package com.ndu.passwordstorage.data

import com.ndu.passwordstorage.model.PasswordEntry

interface PasswordDatabase {
    fun select(): List<PasswordEntry>
    fun insert(passwordEntry: PasswordEntry): Boolean
    fun update(passwordEntry: PasswordEntry): Boolean
    fun delete(passwordEntry: PasswordEntry): Boolean
    fun closeDatabase()
}
